apps/portal/src/content/docs/components/form.mdx

```diff
@@ -7,63 +7,144 @@ The `Form` component is a container for form elements, providing a structured la
 
 import { DocsPage } from "@/components/docs-page";
 
+<DocsPage.ComponentExampleWithZodAndForm
+  client:only
+  code={`() => {
+  const [submittedValue, setSubmittedValue] = React.useState<FormValues | null>(null);
+
+const inputRef = React.useRef<HTMLInputElement | null>(null);
+
+const formSchema = z.object({
+name: z.string().trim().min(3, "Input must be at least 3 characters"),
+
+    // 🚨 z.coerce.number() - Must be used for number fields
+    // https://github.com/shadcn-ui/ui/issues/421#issuecomment-1561080201
+    age: z.coerce.number().min(1, "Number must be at least 1"),
+
+    comment: z.string().trim().min(10, "Comment must be at least 10 characters"),
+
+});
+
+// Infer the type from the schema
+const FormValues = z.infer<typeof formSchema>;
+
+// Initialize react-hook-form with zod resolver
+const formMethods = useForm<FormValues>({
+resolver: zodResolver(formSchema),
+defaultValues: {
+name: "John doe",
+age: 1,
+comment: "Lorem ipsum",
+},
+});
+
+const { register, handleSubmit } = formMethods;
+
+// Handle form submission
+const onSubmit = (data: FormValues) => {
+setSubmittedValue(data);
+};
+
+React.useEffect(() => {
+if(!formMethods.formState.isValid){
+setSubmittedValue(null);
+}
+},[formMethods.formState.isValid]);
+
+return (
+
+<div className="w-[500px]">
+{/* ‼️ It is mandatory to pass all return values from useForm to FormWrapper */}
+<FormWrapper {...formMethods} onSubmit={handleSubmit(onSubmit)}>
+<FormInput.Text
+label="Input Field"
+placeholder="Enter at least 3 characters"
+{...register("name")}
+/>
+
+        <FormInput.Number
+          label="Number Field"
+          placeholder="Enter a number"
+          {...register("age")}
+        />
+
+        <FormInput.Textarea
+          label="Textarea Field"
+          placeholder="Enter a textarea"
+          {...register("comment")}
+        />
+
+        <Button type="submit" disabled={formMethods.formState.isSubmitting}>
+          {formMethods.formState.isSubmitting ? "Submitting..." : "Submit"}
+        </Button>
+
+        <h4>Submitted Values</h4>
+        {formMethods.formState.isValid && submittedValue && (
+          <pre className="mt-4 p-3 bg-cn-background-3 text-cn-foreground-1 rounded whitespace-break-spaces text-sm">
+            {JSON.stringify(submittedValue, null, 2)}
+          </pre>
+        )}
+      </FormWrapper>
+    </div>
+
+);
+}`}
+/>
+
+### Form with Zod and React Hook Form
+
 ```tsx
```

Comment:
This works, but can we stick to `typescript jsx` for consistency with other docs?